xla/service/algebraic_simplifier.cc

```diff
@@ -4569,6 +4569,13 @@ absl::Status AlgebraicSimplifierVisitor::HandleMultiply(
         HloInstruction::CreateUnary(multiply->shape(), HloOpcode::kExp, add));
   }
 
+  VLOG(10) << "trying transform [sqrt(x) * sqrt(x) => x], for x >= 0 "
+           << multiply->ToString();
+  if (Match(multiply, m::Multiply(m::Sqrt(m::Op(&a)), m::Sqrt(m::Op(&a)))) &&
```

Comment:
This line has the same problem as in your other PR, you don't verify that a is the same. Can you please send a followup PR to fix this?

Reply:
Actually I found that the same issue already exists just below (since 2019), so I am going to fix this myself.